File: src/model/Conveyor.java
package model;

import java.util.*;

public class Conveyor {

    private final static int BELT_LENGTH = 7;

    private Queue<Biscuit> slots;

    public Conveyor() {
        this.slots = new ArrayDeque<>();
    }

    public Optional<Biscuit> turn() {

        Optional<Biscuit> biscuit = Optional.empty();
        if (this.slots.size() == BELT_LENGTH) {
            biscuit = Optional.ofNullable(this.slots.remove());
        }

        return biscuit;
    }

    public Optional<Biscuit> turnOff() {

        Optional<Biscuit> biscuit = Optional.empty();

        if (!this.slots.isEmpty()) {
            if (this.slots.peek().getBiscuitState() != BiscuitState.RAW) {
                biscuit = Optional.ofNullable(this.slots.remove());
            }
        }

        return biscuit;
    }

    public void addBiscuit(Biscuit biscuit) {
        this.slots.add(biscuit);
    }

    public boolean hasBiscuitToStamp() {
        return this.slots.size() > 1;
    }
}

File: js-solution/components/conveyor.js
import { queue } from "../utils/queue.js";

const conveyor = (size) => {

    let slots = queue();
    const beltLength = size;

    return {
        add: (item) => {
            slots.add(item);
        },
        turnOn: () => {
            if (slots.size() == beltLength) {
                return slots.remove();
            }

            return null;
        },
        turnOff: () => {
            if (!slots.isEmpty()) {
                return slots.remove();
            }

            return null;
        },
        getSlotOccupancy: () => {
            return slots.size();
        }
    }
};

export { conveyor };